test/index.spec.ts

```diff
@@ -1,5 +1,80 @@
-describe('server checks', () => {
-  it('server is created without error', () => {
-    console.log('Test passed');
+import * as dotenv from 'dotenv';
+dotenv.config({ path: `${__dirname}/../test.env` });
+
+import axios from 'axios';
+import { expect } from 'chai';
+import { initApolloServer } from '../src/apollo-server';
+import { AppDataSource } from '../src/data-source';
+import { User } from '../src/entity/User';
+import { comparePassword } from '../src/encryptPassword';
+
+const connection = axios.create({ baseURL: `http://localhost:${process.env.APOLLO_SERVER_PORT}` });
+
+before(async () => {
+  await AppDataSource.initialize();
+  await initApolloServer();
+});
+
+describe('Users Query Test', () => {
+  it('Should bring number of users from database if server is online', async () => {
+    const query = `query User{
+      users{
+        id
+        name
+        email
+        birthdate
+      }
+    }`;
+
+    const result = await connection.post('/graphql', { query });
+    expect(result.data.data.users.length).to.be.a('number');
+    // console.log(JSON.stringify(result.data.data.users));
+  });
+});
+
+describe('CreateUser Mutation Test', () => {
+  it('Should result have the same data that was informed in the input', async () => {
+    const query = `mutation CreateUser($input: UserInput){
+      createUser(input: $input) {
+        id
+        name
+        email
+        birthdate
+      }
+    }`;
+
+    const input = {
+      name: 'MochaUser',
+      email: 'mochauser@email.com',
+      password: 'mochauserPassword1',
+      birthdate: '01-01-1900',
+    };
+
+    const result = await connection.post('/graphql', {
+      query,
+      variables: { input },
+    });
+
+    const createdUser = result.data.data.createUser;
+    const userInDatabase = (await AppDataSource.getRepository('user').findOneBy({ email: input.email })) as User;
```

Comment:
```suggestion
    const userInDatabase = await AppDataSource.getRepository(User).findOneBy({ email: input.email });
```

Se vc passar a classe pro `getRepository`, provavelmente não vai precisar do cast depois.

Reply:
Feito alterações.